packages/react-instantsearch-hooks-dom/src/ui/Highlight.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { Fragment } from 'react';
+
+import { cx } from './lib/cx';
+
+type HighlightPartProps = {
+  children: React.ReactNode;
+  baseClassName: string;
+  highlightedTagName: React.ReactType;
+  nonHighlightedTagName: React.ReactType;
+  isHighlighted: boolean;
+};
+
+function HighlightPart({
+  baseClassName,
+  children,
+  highlightedTagName,
+  isHighlighted,
+  nonHighlightedTagName,
+}: HighlightPartProps) {
+  const TagName = isHighlighted ? highlightedTagName : nonHighlightedTagName;
+
+  return (
+    <TagName
+      className={
+        isHighlighted
+          ? `${baseClassName}-highlighted`
+          : `${baseClassName}-nonHighlighted`
+      }
+    >
+      {children}
+    </TagName>
+  );
+}
+
+type HighlightedPart = {
+  isHighlighted: boolean;
+  value: string;
+};
+
+export type HighlightProps = React.HTMLAttributes<HTMLSpanElement> & {
```

Comment:
Will `React.HTMLAttributes<HTMLSpanElement>` work with any tag name the user would provide?

Reply:
True, I got mixed up between types.